### PR TITLE
Fix intermittent issue with appendonly regression test

### PR DESCRIPTION
**Pull request by d**
A `CREATE TABLE AS` without a `DISTRIBUTED BY` clause will create a
randomly distributed table, when optimized by ORCA. The plan for the
CTAS will have a redistribute motion (random) between the scan and the
insert. Depending on your data, this style of plan could be more even,
equally even, or less even than a hash distributed table (the kind of
distribution usually assumed by planner).

This commit changes the test to explicitly distribute by the same column
that planner would guess.

Co-authored-by: Jesse Zhang <sbjesse@gmail.com>

### CODE REVIEW

**Comment by ashwinstar** [comment]
Thank You!

**Comment by ashwinstar** [comment]
@d this mostly needs to be ported to 5X_STABLE as well.

**Comment by d** [comment]
Back ported to 5X_STABLE at commit 9882f9c

**Comment by d** [comment]
Jimmy, Ashwin and I talked offline about how often the test would fail before this patch. Posting here for posterity:

## TL;DR
If we haven't done that `DISTRIBUTED BY` commit, there's at least a 24% chance that segment 0 gets no more than 2399 tuples ( [I_(2/3)(6701, 3300) = 0.237](http://www.wolframalpha.com/input/?i=BetaRegularized%5B2%2F3,+6701,+3300%5D) where I_z(a,b) is the [regularized incomplete beta function](http://mathworld.wolfram.com/RegularizedBetaFunction.html).
 

Long version:
The number of tuples a segment gets in a `DISTRIBUTED RANDOMLY` table is a binomial distribution, so we can directly apply the cumulative distribution function with k = 3299, p = 1/3, n = 10000 to get the precise probability: https://en.wikipedia.org/wiki/Binomial_distribution#Cumulative_distribution_function
